File: modules/angular2/src/core/util/decorators.dart
library angular2.core.util.decorators;

import 'dart:mirrors';

class ReflectRegistry {
  List<Type> getForAnnotation(Type annotation) {
    List<Type> list = new List<Type>();
    currentMirrorSystem().libraries.forEach((uri, lib) {
      lib.declarations.forEach((s, decl) {
        decl.metadata.where((m) => m.type.reflectedType == annotation).forEach((m) {
          list.add(decl.reflectedType);
        });
      });
    });
    return list;
  }
}

ReflectRegistry reflectRegistry = new ReflectRegistry();

File: npm/angular2/ts/src/alt_router/metadata/decorators.ts
import {RoutesMetadata, RouteMetadata} from "./metadata";
import {makeDecorator} from 'angular2/src/core/util/decorators';
export interface RoutesFactory {
  (routes: RouteMetadata[]): any;
  new (routes: RouteMetadata[]): RoutesMetadata;
}
export var Routes: RoutesFactory = <RoutesFactory>makeDecorator(RoutesMetadata);
